fili-core/src/main/java/com/yahoo/bard/webservice/web/filters/RoleBasedAuthFilter.java

```diff
@@ -50,17 +51,53 @@ public RoleBasedAuthFilter(ObjectMapper mapper) {
     @Override
     @SuppressWarnings("checkstyle:cyclomaticcomplexity")
     public void filter(ContainerRequestContext containerRequestContext) throws IOException {
-        List<String> allowedUserRoles = getAllowedUserRoles();
-        if (!allowedUserRoles.isEmpty() && !isUserInRole(allowedUserRoles, containerRequestContext)) {
-                abortRequest(
-                        containerRequestContext,
-                        Response.Status.FORBIDDEN,
-                        BouncerAuthorizationStatus.ACCESS_DENIED
-                );
+
+        if (!isAuthorized(containerRequestContext)) {
+            abortRequest(
+                    containerRequestContext,
+                    Response.Status.FORBIDDEN,
+                    BouncerAuthorizationStatus.ACCESS_DENIED
+            );
         }
     }
 
     /**
+     * Request authorization logic, bypassing OPTIONS request and authorize if user is in authorized role or
+     * authorized role is not specified.
+     *
+     * @param containerRequestContext containing the request context information
+     *
+     * @return true if request is authorized, false otherwise
+     */
+    private boolean isAuthorized(ContainerRequestContext containerRequestContext) {
+        return isBypassMethod(containerRequestContext) || isUserInRole(getAllowedUserRoles(), containerRequestContext);
+    }
+
+    /**
+     * Returns true to bypass the filter if the request is a Http OPTIONS method to allow CORS by default.
+     *
+     * @param containerRequestContext Request context containing request method information
+     *
+     * @return true to bypass the filter, and false otherwise
+     */
+    protected boolean isBypassMethod(ContainerRequestContext containerRequestContext) {
+        return containerRequestContext.getMethod().equals(HttpMethod.OPTIONS);
+    }
+
+    /**
+     * Checks if an user role belongs to the list of allowed roles.
+     *
+     * @param allowedUserRoles List of allowed roles
+     * @param containerRequestContext Request context contains the user role information
+     *
+     * @return boolean based on the user role
+     */
+    protected boolean isUserInRole (List<String> allowedUserRoles, ContainerRequestContext containerRequestContext) {
+        SecurityContext securityContext = containerRequestContext.getSecurityContext();
+        return allowedUserRoles.isEmpty() ? true : allowedUserRoles.stream().anyMatch(securityContext::isUserInRole);
```

Comment:
Can simplify this using `||` like so: `allowedUserRoles.isEmpty() || allowedUserRoles.stream().anyMatch(securityContext::isUserInRole);`